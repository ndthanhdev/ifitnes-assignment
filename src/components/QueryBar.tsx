import React from 'react'
import { InputBase, Paper } from '@material-ui/core'
import { makeStyles } from '@material-ui/styles'
import _ from 'lodash'

const useStyles = makeStyles({
  root: {
    display: 'flex',
  },
  inputBase: {
    flex: 1,
    marginLeft: '0.5rem',
    height: '3rem',
  },
  input: {
    textTransform: 'uppercase',
  },
  iconButton: {
    padding: '0.5rem',
  },
})

export const QueryBar: React.FC<{
  onChange?: (value: string) => any
}> = ({ onChange = () => {} }) => {
  const classes = useStyles()
  const handleChange = _.debounce(onChange, 500)

  return (
    <Paper className={classes.root}>
      <InputBase
        placeholder="Start typing your queries ..."
        className={classes.inputBase}
        onChange={e => handleChange(e.target.value)}
        inputProps={{ className: classes.input }}
      />
    </Paper>
  )
}
