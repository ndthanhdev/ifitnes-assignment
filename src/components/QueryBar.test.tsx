import { create } from 'react-test-renderer'
import { QueryBar } from './QueryBar'
import React from 'react'

it('render', () => {
  const component = create(<QueryBar />)

  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})
