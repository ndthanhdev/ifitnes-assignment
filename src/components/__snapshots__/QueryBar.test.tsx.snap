// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded Hook-root-vy1bts"
>
  <div
    className="MuiInputBase-root Hook-inputBase-m0z6e8"
    onClick={[Function]}
  >
    <input
      className="MuiInputBase-input Hook-input-10yzuzw"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Start typing your queries ..."
      type="text"
    />
  </div>
</div>
`;
